### PR TITLE
dask: `Data.HDF_chunks`

### PR DESCRIPTION
**Pull request by davidhassell**
No dask implications here, just tidying up something that never really got finished from way back.

### CODE REVIEW

**Comment by davidhassell** [comment]
Hi Sadie, I've just noticed that we have `nc_clear_hdf5_chunksizes`, `nc_hdf5_chunksizes`, and `nc_set_hdf5_chunksizes` inherited from cfdm, which might make `HDF_chunks` redundant. Please don't look at this yet while I see if that's the case. Thanks.

**Comment by davidhassell** [comment]
OK - using the cfdm methods is the way to go, as it's only `cfdm.NetCDFWrite` that knows how to dealy with the HDF chunk sizes.

**Comment by davidhassell** [comment]
I've left some uses of `HDF_chunks` in, as they are in LAMA-based serialisation methods that are not daskified yet and will need a complete overhaul anyway.

File: cf/data/data.py [hunk 1]
```diff
@@ -10965,41 +10941,85 @@ def flip(self, axes=None, inplace=False, i=False):
...
-            self._HDF_chunks = None
-            return org_HDF_chunks
+            chunks: `dict` or `None`, *optional*
+                Specify HDF chunk sizes.
 
-        axes = self._axes
-        for axis, size in chunks.items():
-            _HDF_chunks[axes[axis]] = size
+                When no positional argument is provided, the HDF chunk
+                sizes are unchanged
```
Comment (sadielbartholomew) [inline cf/data/data.py:10964]:
```suggestion
                sizes are unchanged.
```

**Review by sadielbartholomew — APPROVED** [review]
All good (and very dedicated of you to have added new docstring examples to a deprecated method)!

File: cf/data/data.py (hunk 1 above)
Comment (davidhassell) [reply cf/data/data.py]: ... likewise for typos in deprecated methods. Someone somewhere might will be glad for our efforts!

**Comment by davidhassell** [comment]
Just checking - is good to merge? Thanks.

**Comment by sadielbartholomew** [comment]
> Just checking - is good to merge? Thanks.

Yes, please go ahead.

**Pull request merged**